src/hotspot/share/ci/ciEnv.hpp

```diff
@@ -191,15 +191,13 @@ class ciEnv : StackObj {
     if (o == NULL) {
       return NULL;
     } else {
-#ifndef PRODUCT
       if (ReplayCompiles && o->is_klass()) {
         Klass* k = (Klass*)o;
         if (k->is_instance_klass() && ciReplay::is_klass_unresolved((InstanceKlass*)k)) {
           // Klass was unresolved at replay dump time. Simulate this case.
           return ciEnv::_unloaded_ciinstance_klass;
         }
       }
```

Comment:
May be make this part of code as `ciReplay` method and place it into ciReplay.cpp to reduce code in .hpp file.

Reply:
@vnkozlov , I moved the code into `ciObjectFactory::get_metadata()` since it seemed to fit there very well. Is that okay or do you really want it to live in ciReplay.cpp?